include/lbann/execution_algorithms/kfac/kfac_block.hpp

```diff
@@ -114,6 +140,24 @@ class kfac_block {
     LBANN_ERROR("this function should be called via a sub-class.");
   }
 
+  virtual void start_communication_forward_end(
+      lbann_comm* comm){
+    LBANN_ERROR("this function should be called via a sub-class.");
+  }
+  virtual void end_communication_forward_end(
+      lbann_comm* comm){
+    LBANN_ERROR("this function should be called via a sub-class.");
+  }
+  virtual void start_communication_backward_end(
+      lbann_comm* comm){
+    LBANN_ERROR("this function should be called via a sub-class.");
+  }
+  virtual void end_communication_backward_end(
+      lbann_comm* comm){
+    LBANN_ERROR("this function should be called via a sub-class.");
+  }
```

Comment:
```suggestion
  virtual void start_communication_forward_end(lbann_comm* comm) = 0;
  virtual void end_communication_forward_end(lbann_comm* comm) = 0;
  virtual void start_communication_backward_end(lbann_comm* comm) = 0;
  virtual void end_communication_backward_end(lbann_comm* comm) = 0;
```
Is there a reason these aren't left pure virtual??